contracts/wormhole/contracts/mapping-token/v2/MappingTokenFactory.sol

```diff
@@ -241,133 +197,22 @@ contract MappingTokenFactory is Initializable, Ownable, DailyLimit, ICrossChainF
-            // todo we should transform this messageId to bridged outboundLane messageId
-            uint256 messageId = IInboundLane(msg.sender).encodeMessageKey(inboundLaneNonce.last_delivered_nonce);
-            IGuard(guard).deposit(messageId, mappingToken, recipient, amount);
-        } else {
-            IERC20(mappingToken).mint(recipient, amount);
-        }
-    }
-
-    /**
-     * @notice burn mapping token and unlock remote original token, waiting for the confirm
-     * @param bridgedLanePosition the bridged lane position to send the unlock message
-     * @param mappingToken the burt mapping token address
-     * @param recipient the recipient of the remote unlocked token
-     * @param amount the amount of the burn and unlock
-     */
-    function burnAndRemoteUnlockWaitingConfirm(
+    function _sendMessage(
+        uint32 bridgedChainPosition,
         uint32 bridgedLanePosition,
-        address mappingToken,
-        address recipient,
-        uint256 amount
-    ) external payable whenNotPaused {
-        require(amount > 0, "MappingTokenFactory:can not transfer amount zero");
-        OriginalInfo memory info = mappingToken2OriginalInfo[mappingToken];
-        require(info.originalToken != address(0), "MappingTokenFactory:token is not created by factory");
-        // Lock the fund in this before message on remote backing chain get dispatched successfully and burn finally
-        // If remote backing chain unlock the origin token successfully, then this fund will be burned.
-        // Otherwise, this fund will be transfered back to the msg.sender.
-        require(IERC20(mappingToken).transferFrom(msg.sender, address(this), amount), "MappingTokenFactory:transfer token failed");
-
-        uint256 outBoundId = encodeBridgedBoundId(info.bridgedChainPosition, bridgedLanePosition);
+        address receiver,
+        bytes memory message
+    ) internal returns(uint256) {
+        uint256 outBoundId = encodeBridgedBoundId(bridgedChainPosition, bridgedLanePosition);
         address outboundLane = outboundLanes[outBoundId];
         require(outboundLane != address(0), "MappingTokenFactory:the outbound lane is not exist");
-        bytes memory unlockFromRemote = abi.encodeWithSelector(
-            IBacking.unlockFromRemote.selector,
-            address(this),
-            info.originalToken,
-            recipient,
-            amount
-        );
         uint256 fee = IFeeMarket(feeMarket).market_fee();
         require(msg.value >= fee, "MappingTokenFactory:not enough fee to pay");
-        uint256 messageId = IOutboundLane(outboundLane).send_message{value: fee}(info.backingAddress, unlockFromRemote);
-        unlockRemoteUnconfirmed[messageId] = UnconfirmedInfo(msg.sender, mappingToken, amount);
+        uint256 messageId = IOutboundLane(outboundLane).send_message{value: fee}(receiver, message);
         if (msg.value > fee) {
             payable(msg.sender).transfer(msg.value - fee);
```

Comment:
The same line in backing is using `msg.value.sub(fee)`

Reply:
The new version, this is moved to the `HelixApp`, and we don't need to use safe.sub since we have checked the value at the above line.